bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/internal/rules/YamlRuleDTO.java

```diff
@@ -0,0 +1,396 @@
+                YamlConditionDTO condition;
+                for (Iterator<JsonNode> iterator = partial.conditions.elements(); iterator.hasNext();) {
+                    conditionNode = iterator.next();
+                    condition = mapper.treeToValue(conditionNode, YamlConditionDTO.class);
+                    condition.type = ModuleTypeAliases.aliasToType(Condition.class, condition.type);
+                    if ((config = condition.config) != null && config.containsKey("script")
+                            && config.get("type") instanceof String type) {
+                        if (!type.equals(translatedType = MIMETypeAliases.aliasToType(type))) {
+                            config.put("type", translatedType);
+                        }
+                    }
+                    conditions.add(condition);
+                }
+                result.conditions = conditions;
+            }
+            if (partial.triggers != null && !partial.triggers.isEmpty()) {
+                if (!partial.triggers.isArray()) {
+                    throw new SerializationException("Expected triggers to be an array node");
+                }
+                List<YamlModuleDTO> triggers = new ArrayList<>(partial.triggers.size());
+                JsonNode triggerNode;
+                YamlModuleDTO trigger;
+                for (Iterator<JsonNode> iterator = partial.triggers.elements(); iterator.hasNext();) {
+                    triggerNode = iterator.next();
+                    trigger = mapper.treeToValue(triggerNode, YamlModuleDTO.class);
+                    trigger.type = ModuleTypeAliases.aliasToType(Trigger.class, trigger.type);
+                    if ((config = trigger.config) != null && config.containsKey("script")
+                            && config.get("type") instanceof String type) {
+                        if (!type.equals(translatedType = MIMETypeAliases.aliasToType(type))) {
+                            config.put("type", translatedType);
+                        }
+                    }
+                    triggers.add(trigger);
+                }
+                result.triggers = triggers;
+            }
+        } catch (JsonProcessingException | IllegalArgumentException e) {
+            throw new SerializationException(e.getMessage(), e);
+        }
+        return result;
+    }
+
+    @Override
+    public YamlElement cloneWithoutId() {
+        YamlRuleDTO copy;
+        try {
+            copy = (YamlRuleDTO) super.clone();
+            copy.uid = null;
+            return copy;
+        } catch (CloneNotSupportedException e) {
+            // Will never happen
+            return new YamlRuleDTO();
+        }
+    }
+
+    @Override
+    public boolean isValid(@Nullable List<@NonNull String> errors, @Nullable List<@NonNull String> warnings) {
+        // Check that uid is present
+        if (uid == null || uid.isBlank()) {
+            addToList(errors, "invalid rule: uid is missing");
```

Comment:
For better consiste,cy with items and things, I am proposing: invalid rule: uid is missing **while mandatory**

Reply:
Same remark is applicable for YamlRuleTemplateDTO